### PR TITLE
#1218 Finance Config 

### LINKED ISSUE #1218: Finance - Finance Team Config (Admin Tools)
### Description

We need to allow the finance team to add some items to the database. we will allow them to do this through the admin tools section of the website

### Acceptance Criteria

1. The "Account Code" column in the "Account Codes" table should be editable
2. In the new/edit reimbursement request form, only account codes marked allowed should appear
4. Only vendors in the "Registered Vendors" list should appear in the Purchased From field in the new/edit reimbursement request form

### Proposed Solution

^

### Mocks

image

epic #493

### PR DESCRIPTION
**Pull request by walker-sean**
## Changes

The frontend stuff is set up

## Notes

Need some endpoints done to add create/edit functionality

## Screenshots

image
image

### account code edit endpoint screenshots:

**Get expense type with certain reimbursement request id**
#1218-getExpenseType-test

**Send edit expense type code request with new account code number and name** 
#1218-postExpenseType-ss

**Get updated expense type after POST request**
#1218-getExpenseTypeAfter

## To Do

- [ ] Endpoints for editing vendors and expense types
- [ ] All hooks
- [ ] Integrating edit and create functionality
- [ ] Finish connecting the edit account code page to endpoint

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1218

### CODE REVIEW

**Comment by callmejim1226** [comment]
I will be working on the expense type edit endpoint and push the work on this PR by tmr or soon.

File: src/frontend/src/pages/AdminToolsPage/AdminToolsPage.tsx [hunk 1]
```diff
@@ -8,14 +8,16 @@ import AdminToolsSlackUpcomingDeadlines from './AdminToolsSlackUpcomingDeadlines
...
 
 const AdminToolsPage: React.FC = () => {
   const currentUser = useCurrentUser();
 
   return (
     <PageLayout title="Admin Tools">
       <AdminToolsUserManagement />
-      {isAdmin(currentUser.role) && <AdminToolsSlackUpcomingDeadlines />};
+      {isAdmin(currentUser.role) && <AdminToolsSlackUpcomingDeadlines />}
+      <AdminToolsFinanceConfig />
```
Comment (RChandler234) [inline src/frontend/src/pages/AdminToolsPage/AdminToolsPage.tsx:20]: let's restrict this so only admins can see it

File: src/frontend/src/pages/AdminToolsPage/AdminToolsFinanceConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+          <TableBody>{accountCodesTableRows}</TableBody>
+        </Table>
+      </TableContainer>
+    </Grid>
+  );
+
+  return (
+    <PageBlock title="Finance Config">
+      <Grid container spacing="3%">
+        {registeredVendors}
```
Comment (RChandler234) [inline src/frontend/src/pages/AdminToolsPage/AdminToolsFinanceConfig.tsx:141]: you can put the vendors table and the accountCodes table in their own files

File: src/frontend/src/pages/AdminToolsPage/AdminToolsFinanceConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                Date Registered
+              </TableCell>
+              <TableCell align="left" sx={{ fontSize: '16px', fontWeight: 600, border: '1px solid black' }}>
+                Vendor Name
+              </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {vendorTableRows}
+            {newVendorRow}
```
Comment (RChandler234) [inline src/frontend/src/pages/AdminToolsPage/AdminToolsFinanceConfig.tsx:92]: this is tricky cause we need to be able to create new expense types, new account codes, and be able to edit account codes. I think doing it inline will be pretty tricky, for the mvp, I'd suggest making a modal pop up when you click on a row and then allow the user to make edits in there. 

Same for creating, I think buttons right under both of the tables that pops open a modal for creating a vendor or account code with inputs for all the data a user is inputting will be easier

Lmk if you have questions ab any of that

**Review by RChandler234 — CHANGES_REQUESTED** [review]
image
fix the gaps in the corners

File: src/frontend/src/pages/AdminToolsPage/EditAccountCodeView.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  onHide: () => void;
+  onSubmit: () => Promise<void>;
+  accountCode: ExpenseType;
+}
+
+const EditAccountCodeView: React.FC<EditAccountCodeViewProps> = ({
+  showModal,
+  onHide,
+  onSubmit,
+  accountCode
```
Comment (callmejim1226) [inline src/frontend/src/pages/AdminToolsPage/EditAccountCodeView.tsx:18]: I'm having an issue with updating accountCode.code variable after the user submits this modal with a new code variable. I wonder if we should use useState hook to get updated account code. Let me know if we have other way to get a new account number after user clicks submit button in the modal.

**Review by RChandler234 — APPROVED** [review]

**Pull request merged**